scripts/cmake/common.cmake

```diff
@@ -72,6 +72,25 @@ function(doctest_add_test)
     doctest_add_test_impl(${ARGN} JUNIT_OUTPUT)
 endfunction()
 
+# In order not to miss subcases, parse a file with regex to find them
+function(doctest_detect_test_cases FILE_NAME OUTPUT_NAME)
```

Comment:
Not sure if relevant, but just wanted to post about `doctest_discover_tests` here in case you're not aware of it (I haven't used it myself):
https://github.com/doctest/doctest/blob/master/scripts/cmake/doctest.cmake#L108